docs/docs-beta/docs/dagster-plus/deployment/code-locations.md

```diff
@@ -3,4 +3,237 @@ title: "Code locations"
+- Adding a new code location to a Git monorepo
+
+<details>
+<summary>Prerequisites</summary>
+
+1. You should have a Dagster project, we recommend setting this project up as a Python module following our [recommended project structure](/tutorial/create-new-project).
+
+2. You should know whether or not you are using Dagster Plus Serverless or Dagster Plus Hybrid. Your administrator will know that, or you can find the agent type by navigating to your Dagster+ organization, clicking on the "Deployment" page, and selecting the "Agent" tab. Dagster+ Serverless organizations will have a label "Managed by Dagster+".
+
+</details>
+
+Adding a code location follows two general steps:
+- Putting your Dagster code in a place that's accessible by your agent. For Dagster+ Serverless you can skip this step. For Dagster+ Hybrid this usually means building a Docker image with your code and pushing it to a registry.
+- Notifying Dagster+ of the new or updated code location. This will be done by using the Dagster+ Python client.
+
+Often these two steps are handled by CICD connected to your Git repository.
+
+
+## Add a new code location manually
+
+Start by installing the `dagster-cloud` Python client:
+
+```
+pip install dagster-cloud
+```
+
+Next you will want need to authenticate this Python client:
+
+1. Navigate to your Dagster+ organization. Click on your user icon in the upper right corner, then select "Organization settings". Select "Tokens" and press "Create user token". Copy the token.
+
+2. Set the following environment variables:
+
+```bash
+export  DAGSTER_CLOUD_ORGANIZATION="organization-name" # if your URL is https://acme.dagster.plus your organization name is "acme"
+export  DAGSTER_CLOUD_API_TOKEN="your-token"
+```
+
+3. Add the code location. The following example assumes you are running the command from the top-level working directory of your Dagster project with a project named "quickstart" structured as a Python module named "quickstart".
+
+```bash
+$pwd
+/projects/quickstart
+$tree
+/quickstart
+    setup.py
+    pyproject.toml
+    /quickstart
+        __init__.py
+        /assets
+        /resources
+```
+
+If you are using Dagster+ Serverless, you'll want to run:
+
+```bash
+dagster-cloud serverless deploy-python-executable --deployment prod --location-name quickstart --module-name quickstart
+```
+
+Use `dagster-cloud serverless deploy-python-executable --help` to see all available options. Running the command multiple times with the same location name will *update* the code location. Running the command with a new location name will *add* a code location.
+
+If you are using Dagster+ Hybrid, make sure you have deployed the code appropriately by building a Docker image and pushing it to a Git repository. Then run:
+
+```bash
+dagster-cloud deployment add-location --deployment prod --location-name quickstart --module-name quickstart --image 764506304434.dkr.ecr.us-west-2.amazonaws.com/hooli-data-science-prod:latest
+```
```

Comment:
I think putting these instructions into tabs will be helpful in reducing noise for folks who only care about one deployment type. Let me know if you want to take a swing, otherwise I'll do it on a follow up.

Reply:
Additionally, how should the user confirm that everything worked correctly?

Reply:
I wonder if there's a way to make all the tabs respond to a single click (so if you click Hybrid..they all become Hybrid. I'm sure the answer is yes)

Reply:
ah cool. I'm positive we can create a toggle on the docs somewhere that sets this value throughout the site. Maybe file a ticket for post-sprint?